ocaml/idl/ocaml_backend/gen_api.ml

```diff
@@ -82,10 +82,19 @@ let gen_record_type ~with_module highapi tys =
         sprintf "\"%s\",rpc_of_%s x.%s" (String.concat "_" fld.DT.full_name)
           (OU.alias_of_ty fld.DT.ty) (OU.ocaml_of_record_field (obj_name :: fld.DT.full_name))
       in
-
+      let get_default fld =
+        let default_value =
+          match fld.DT.ty with
+          | DT.Set (DT.Ref _) -> Some (DT.VSet [])
+          | _ -> fld.DT.default_value
+        in
+        match default_value with
+          None -> "None"
+        | Some default -> sprintf "(Some (%s))" (Datamodel_values.to_ocaml_string ~v2:true default)
+      in
       let make_to_field fld =
-        sprintf "%s = %s_of_rpc (List.assoc \"%s\" x)" (field fld) (OU.alias_of_ty fld.DT.ty)
-          (String.concat "_" fld.DT.full_name)
+        sprintf {|%s = %s_of_rpc (assocer "%s" x %s)|} (field fld) (OU.alias_of_ty fld.DT.ty)
+          (String.concat "_" fld.DT.full_name) (get_default fld)
```

Comment:
It is not necessary to generate these `X_of_rpc`, and `rpc_of_X` things, rpclib can do it for us. I have a set of working patches, one of this removes these custom functions and moves to ones generated by rpclib:
https://github.com/gaborigloi/xen-api/commit/2c4fc73cf72c192feaaf091c80c78534a7729fd2

Reply:
Not in this case, we need to customise them and replace the autogenerated ones.

Reply:
Maybe we can use the new `[@default ...]` key but it needs more work to serialize it correctly and it needs further testing. It is an improvement that can be added later and we should keep in mind to try it